### ISSUE TITLE
Extended directives (actually BOM mark) not ignored when parsing M3U playlists.

### ISSUE THREAD

**Issue by ianmacd** [henkelis/sonospy#55]
Scanning .m3u playlists treats the following as if they were tracks:
# EXTM3U - header - must be first line of file
# EXTINF - extra info - length (seconds), title

**Comment by henkelis** [comment]
Is the # the first character of the line? Can you send me an example file please.

**Comment by ianmacd** [comment]
I actually deleted the one I had that caused the problem, because it was a spurious file, but there's a documented example here:

http://en.wikipedia.org/wiki/M3U#Example

**Comment by henkelis** [comment]
I have ones that work correctly here, I wanted to find out why yours doesn't.

The code ignores all lines starting with #.

**Comment by ianmacd** [comment]
I've recovered the file from a back-up, but don't appear able to attach it here. I'll post it to the list.

**Comment by ianmacd** [comment]
I turns out that it contains the BOM (0xfeff) mark.

http://en.wikipedia.org/wiki/UTF-8#Byte_order_mark

I guess this will be a more generic problem when handling files created by stupid software.

**Comment by henkelis** [comment]
Indeed. I created a playlist with mp3tag under Wine, and it created it 
as a Windows file which scan on Linux didn't like.

I'll look at how best to avoid the obvious issues.
